src/shared/container/providers/CacheProvider/implementations/RedisCacheProvider.ts

```diff
@@ -0,0 +1,40 @@
+import Redis, { Redis as IRedis } from 'ioredis'
+import cacheConfig from '@config/cache'
+import ICacheProvider from '../models/ICacheProvider'
+
+export default class RedisCachePRovider implements ICacheProvider {
+  private client: IRedis
+
+  constructor() {
+    this.client = new Redis(cacheConfig.config.redis)
+  }
+
+  public async save(key: string, value: any): Promise<void> {
+    await this.client.set(key, JSON.stringify(value))
+  }
```

Comment:
É possível repassar a Promise para a stack de execução no lugar de executar o trecho dentro de `save`:

```suggestion
  public save(key: string, value: any): Promise<void> {
    return this.client.set(key, JSON.stringify(value))
  }
```